test/test/publisher_subscriber/PublisherSubscriberInteraction.cpp

```diff
@@ -173,6 +173,21 @@ TEST_P(PublisherSubscriberNoLost, PubSub1ContinousFragmentedTopic)
     publisher_.publish(1, 0x80, 1, message);
 }
 
+TEST_P(PublisherSubscriberNoLost, PubSub1WithPing)
+{
+    size_t message_number = 1;
+    publisher_.publish(1, 0x80, message_number, SMALL_MESSAGE);
+    subscriber_.request_data(1, 0x80, SMALL_MESSAGE);
+
+    for (size_t i = 0; i < 5; i++)
```

Comment:
As we know that the agent will transmit the topic to the client at some moment in between of the ping answers due to the output message queue, what do you think about doing something like:

```c
int64_t start_time = uxr_millis();

while(uxr_millis() -  start_time < 10000 && subscriber_.get_received_topics() != message_number){
        subscriber_.ping_agent_session();
}

ASSERT_EQ(subscriber_.get_received_topics(), message_number);
```

Reply:
Updated 57e4658c53749a89e5477488e52c3a3771aeab12

Reply:
Cool